### PR TITLE
document common mistakes and misconceptions in a pitfalls article

### PR DESCRIPTION
**Pull request by alcohol**

### CODE REVIEW

**Comment by Seldaek** [comment]
This should be merged into the troubleshooting page IMO to avoid duplicating places people should look for help.

File: doc/articles/troubleshooting.md [hunk 1]
```diff
@@ -74,6 +74,29 @@ This is a list of common pitfalls on using Composer, and how to avoid them.
...
+frequently overlooked:
+
+> **Note:** While this is convenient at times, it should not be how you use
+> packages in the long term because it comes with a technical limitation. The
+> composer.json metadata will still be read from the branch name you specify
+> before the hash. Because of that in some cases it will not be a practical
+> workaround, and you should always try to switch to tagged releases as soon
+> as you can.
+
+There is no simple work-around to this limitation. It is therefor strongly recommended that you do not use it.
```
Comment (cs278) [inline doc/articles/troubleshooting.md]: therefore

**Comment by cs278** [comment]
I wonder if it's worth including an advisory in the output of `validate` for the `#commitish` feature?

**Comment by Seldaek** [comment]
Maybe? :) I don't know if it's over-reaching or not.. I guess there are valid use cases if you control the packages or something.. not sure really.

**Comment by alcohol** [comment]
I believe that was already added at some point?

**Comment by alcohol** [comment]
See #4529

**Comment by cs278** [comment]
Ah, maybe add link to this doc with that message.

**Pull request merged**